webrender/src/prim_store.rs

```diff
@@ -2784,16 +2788,28 @@ impl PrimitiveInstance {
                             }
                         }
                     }
-                    BrushKind::LineDecoration { ref mut handle, style, orientation, wavy_line_thickness, .. } => {
-                        // Work out the device pixel size to be used to cache this line decoration.
+                    BrushKind::LineDecoration { color, ref mut handle, style, orientation, wavy_line_thickness } => {
+                        // Update opacity for this primitive to ensure the correct
+                        // batching parameters are used.
+                        self.opacity.is_opaque = match style {
+                            LineStyle::Solid => color.a == 1.0,
+                            LineStyle::Dotted |
+                            LineStyle::Dashed |
+                            LineStyle::Wavy => false,
+                        };
 
+                        // Work out the device pixel size to be used to cache this line decoration.
                         let size = get_line_decoration_sizes(
                             &prim_local_rect.size,
                             orientation,
                             style,
                             wavy_line_thickness,
                         );
 
+                        if is_chased {
```

Comment:
would a `if cfg!(debug_assertions)` be interesting here as well, to match the other ones ?